### PR TITLE
refactor: error feedback UX

### PR DESCRIPTION
**Pull request by agatha197**
resolves https://github.com/lablup/backend.ai-webui/issues/2546

### TL;DR
Refactored multiple components for better error handling.

### What changed?
- Updated `ContainerRegistryEditorModal`, `ContainerRegistryList`, `KeypairResourcePolicyList`, `ProjectResourcePolicyList`, `ManageAppsModal`, `ManageImageResourceLimitModal` to utilize `App.useApp` for message and modal usage.
- Removed unnecessary useMessage and useModal hook calls.
- Enhanced error message display with detailed error information from responses.

### How to test?
1. Trigger various actions that would invoke message and modal displays, such as creating, modifying or deleting registries, resource policies, etc.
2. Verify the error messages are displayed correctly.

### Why make this change?
To simplify and unify the error handling approach by leveraging `App.useApp` for message and modal displays, reducing code duplication and improving maintainability.

**Checklist:** (if applicable)

- [x] Mention to the original issue
- [ ] Documentation
- [ ] Minium required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Comment by agatha197** [comment]
* **#2548** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2548?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @agatha197 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/UserResourcePolicyList.tsx [hunk 1]
```diff
@@ -187,9 +187,16 @@ const UserResourcePolicyList: React.FC<UserResourcePolicyListProps> = () => {
                   variables: {
                     name: row.name,
                   },
-                  onCompleted: (data, errors) => {
-                    if (errors) {
-                      message.error(errors[0]?.message);
+                  onCompleted: (res, errors) => {
+                    if (!res?.delete_user_resource_policy?.ok) {
+                      message.error(res?.delete_user_resource_policy?.msg);
+                      return;
```
Comment (ironAiken2) [inline react/src/components/UserResourcePolicyList.tsx:193]:
```suggestion
                      message.error(res?.delete_user_resource_policy?.msg);
                      setInFlightResourcePolicyName('');
                      return;
```

Wouldn't it be nice to turn off the loading indicator?

Comment (agatha197) [reply react/src/components/UserResourcePolicyList.tsx:193]: I resolved it by adding a flight checking to loading.

**Review by ironAiken2 — APPROVED** [review]
LGTM!

**Pull request merged**